libraries/Microsoft.Bot.Builder.Dialogs/ObjectPath.cs

```diff
@@ -42,8 +42,9 @@ public static bool TryGetValue<T>(object o, string pathExpression, out T value)
                 return false;
             }
 
+            // if JPath expression
             JToken result = null;
-            if (pathExpression.StartsWith("$"))
+            if (pathExpression.StartsWith("$."))
             {
```

Comment:
Not sure what this is doing but wouldn't you need to do the same for @xxx and #xxx ?

Reply:
This if was intended to catch JPath expressions, but then we introduced "$" as a shortcut to dialog and it got into this if :) Here Tom is trying to explicitly enter this if only when we have a jpath query